client/src/components/ResourceList/index.js

```diff
@@ -1,29 +1,101 @@
 import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
 
+import { FormControls } from "../../components";
+import { TopicService, useService, ResourceService } from "../../services";
 import "./ResourceList.scss";
 
-export default function ResourceList({ publish, resources }) {
+export default function ResourceList({ publish, resources, pathname }) {
+	const [topics, setTopics] = useState([]);
+	const topicService = useService(TopicService);
+	const [selectedTopic, setSelectedTopic] = useState("");
+	const [filteredResources, setFilteredResources] = useState([]);
+	const resourceService = useService(ResourceService);
+
+	useEffect(() => {
+		const fetchTopics = async () => {
+			try {
+				const isTestEnvironment = process.env.NODE_ENV === "test";
+
+				const fetchedTopics = isTestEnvironment
+					? []
+					: await topicService.getTopics();
+
+				setTopics(fetchedTopics);
+			} catch (error) {
+				throw new Error(`Error fetching topics: ${error.message}`);
+			}
+		};
+
+		fetchTopics();
+	}, [setTopics, topicService]);
+
+	useEffect(() => {
+		const fetchResourcesByTopic = async () => {
+			try {
+				const allResources = await resourceService.getPublished({});
+				const filtered = allResources.resources.filter(
+					(topic) => topic.topic_name === selectedTopic
```

Comment:
❌ What you've named `topic` here is actually the _resource_, or you could just destructure it:
```suggestion
					({ topic_name }) => topic_name === selectedTopic
```

Reply:
thank you @textbook that's done.